rust/src/tx_builder.rs

```diff
@@ -473,7 +544,19 @@ impl TransactionBuilder {
             collateral: None,
             required_signers: None,
             network_id: None,
-        })
+        };
+        // we must build a tx with fake data (of correct size) to check the final Transaction size
+        let full_tx = fake_full_tx(self, built)?;
+        let full_tx_size = full_tx.to_bytes().len();
+        if full_tx_size > self.max_tx_size as usize {
+            Err(JsError::from_str(&format!(
+                "Maximum transaction size of {} exceeded. Found: {}",
+                self.max_tx_size,
+                full_tx_size
+            )))
+        } else {
+            Ok(full_tx.body)
```

Comment:
This is similar to returning `Ok(built)`, right? Just looks weird that the result used to be that constructed tx body and now as if it is important for it to be taken from the fake `full_tx`